Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -64,7 +64,11 @@ def check_python_framework_links(lib)
   def check_linkage
     return unless formula.prefix.directory?
     keg = Keg.new(formula.prefix)
-    checker = LinkageChecker.new(keg, formula)
+
+    DatabaseCache.new(:linkage) do |database_cache|
+      checker = LinkageChecker.new(keg, database_cache, formula)
+      checker.flush_cache_and_check_dylibs
```

Comment:
Could consider making an argument to `initialize` that will run `flush_cache_and_check_dylibs` if passed?